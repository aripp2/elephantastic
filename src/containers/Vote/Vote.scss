@import './src/variables.scss';

.vote-section {
  width: 100%;
  font-family: $header-font;
  padding-top: 50px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.voting-btns {
  width: 50%;
  margin-bottom: 20px;
  display: flex;
  justify-content: center;
}

.vote-btn {
  height: 50px;
  width: 200px;
  font-size: 1.5em;
  border: 2px groove black;
  border-radius: 5px;
  margin: auto;
  &:hover {
    color: white;
    background-color: $love-color;
  }
}

.voteImg {
    height: auto; 
    width: auto; 
    max-width: 500px; 
    max-height: 400px;
    margin: auto;
    border-radius: 5px;
}

.fav-btn {
  height: 40px;
  width: 200px;
  font-size: 1.5em;
  border: 2px groove black;
  border-radius: 5px;
  margin-top: 20px; 
  &:hover {
    color: white;
    background-color: $love-color;
  } 
}