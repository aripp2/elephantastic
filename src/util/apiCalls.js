const xKey = 'fdae1377-8b39-4c2b-a87f-38d9ef268562';
const getOptions = {
    method: 'GET',
    headers: {
      'x-api-key': xKey
    }
  };

export const getRandom = async() => {
  const url = 'https://api.thedogapi.com/v1/images/search';
  const response = await fetch(url, getOptions)
  if(!response.ok) {
    throw Error('There was an issue getting a pup, try again later.')
  }
  const random = await response.json();
  return random[0]

}

export const postVote = async (imgId, value) => {
  const url = 'https://api.thedogapi.com/v1/votes';
  const options = {
    method: 'POST',
    body: JSON.stringify({
      image_id: imgId,
      value: value
    }),
    headers: {
      'Content-type': 'application/json',
      'x-api-key': xKey
    }
  }

  const response = await fetch(url, options);
  if(!response.ok) {
    throw Error('Unable to submit your vote.')  
  }
  const votes = await response.json()
  console.log('postVote in apiCalls', votes)
}

export const getBreeds = async () => {
  const url = 'https://api.thedogapi.com/v1/breeds';
  const response = await fetch(url, getOptions);
  if(!response.ok) {
    console.log(response)
  }
  const breeds = await response.json();
  // console.log('getBreeds in apiCalls', breeds)
  return breeds;
}

export const getFavorites = async() => {
  const url = 'https://api.thedogapi.com/v1/favourites';
  const response = await fetch(url, getOptions)
  if(!response.ok) {
    console.log(response)
  }
  const favs = await response.json();
  console.log('favs', favs)
  return favs;
}

export const addFavorite = async(imgId) => {
  const url = 'https://api.thedogapi.com/v1/favourites';
  const options = {
    method: 'POST',
    body: JSON.stringify({
      image_id: imgId
    }),
    headers: {
      'Content-type': 'application/json',
      'x-api-key': xKey
    }
  } 

  const response = await fetch(url, options)
  if(!response.ok) {
    throw Error('Unable to add favorite.')
  }
}

export const deleteFavorite = async(favId) => {
  const url = `https://api.thedogapi.com/v1/favourites/${favId}`
  const options = {
    method: 'DELETE',
    headers: {
      'Content-type': 'application/json',
      'x-api-key': xKey
    }
  }
  const response = await fetch(url, options)
  if(!response.ok) {
    throw Error('Unable to remove this favorite.')
  }
}

export const getBreedImages = async(breedId) => {
  const url = `https://api.TheDogAPI.com/images/search?breed_id=${breedId}`
  const response = await fetch(url, getOptions)
  if(!response.ok) {
    console.log('selected breed response', response)
  }
  const selected = await response.json()
  console.log('selected in api', selected)
  return selected
}





