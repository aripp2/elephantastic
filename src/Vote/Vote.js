import React from 'react';
import { connect } from 'react-redux';
import { postVote } from '../util/apiCalls';
// import { setRandom, throwError } from '../actions';

import './Vote.scss';


const Vote = ({ randomPup, updateRandom }) => {
  const { url, id } = randomPup
  console.log(randomPup)
  return (
    <section>
      <button
        onClick={() => {postVote(id, 1); updateRandom()}}
      >LOVE IT!</button>
      <button
        onClick={() => {postVote(id, 0); updateRandom()}}
      >NOT SO MUCH</button>
      <img className='voteImg' src={url} alt='dog'/>
      <button>Favorite</button>

    </section>
  )
}

export const mapStateToProps = ({ randomPup }) => ({
  randomPup
})

// export const mapDispatchToProps = dispatch => ({
//   setRandom: pup => dispatch(setRandom(pup))
// })

export default connect(mapStateToProps)(Vote);