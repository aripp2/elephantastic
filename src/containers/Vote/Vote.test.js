import React from 'react';
import { shallow } from 'enzyme';
import { Vote, mapStateToProps } from './Vote';
import { postVote } from '../../util/apiCalls';

jest.mock('../../util/apiCalls.js');

  const mockFavs = [
    {
      created_at: "2019-11-01T17:26:47.000Z",
      id: 1095,
      image: {id: "ry8KWgqEQ", url: "https://cdn2.thedogapi.com/images/ry8KWgqEQ.jpg"},
      image_id: "ry8KWgqEQ",
      sub_id: null,
      user_id: "6xrk1n"
    },
    {
      created_at: "2019-11-02T18:32:57.000Z",
      id: 1098,
      image: {id: "edNIMgpSw", url: "https://cdn2.thedogapi.com/images/edNIMgpSw.jpg"},
      image_id: "edNIMgpSw",
      sub_id: null,
      user_id: "6xrk1n"
    }
  ];

  const mockRandomPup = {
    breeds: [],
    height: 720,
    id: "anIClTXyD",
    url: "https://cdn2.thedogapi.com/images/anIClTXyD.jpg",
    width: 350
  };

describe('Vote', () => {
  let wrapper;
  const mockUpdateRandom = jest.fn();
  const mockUpdateFavs = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<Vote 
      randomPup={mockRandomPup}
      favorites={mockFavs}
      updateRandom={mockUpdateRandom}
      updateFavs={mockUpdateFavs}
      />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call postVote with value of 1 and updateRandom when vote-btn is clicked', () => {
    wrapper.find('.vote-btn').at(0).simulate('click');
    expect(postVote).toHaveBeenCalledWith("anIClTXyD", 1);

    expect(mockUpdateRandom).toHaveBeenCalled();
  });

  it('should call postVote with value of 0 and updateRandom when vote-btn is clicked', () => {
    wrapper.find('.vote-btn').at(1).simulate('click');
    expect(postVote).toHaveBeenCalledWith('anIClTXyD', 0);

    expect(mockUpdateRandom).toHaveBeenCalled();
  });

  it('should call updateFavs when fav-but is clicked', () => {
    wrapper.find('.fav-btn').simulate('click');
    expect(mockUpdateFavs).toHaveBeenCalled();
  }); 

});

describe('mapStateToProps', () => {

  it('should return an object of favs array and randomPup obj', () => {
    const mockState = {
      favorites: mockFavs,
      randomPup: mockRandomPup,
      errorMsg: '',
      isLoading: false,
      breeds: []
    };
    const expected = {
      favorites: mockFavs,
      randomPup: mockRandomPup
    }
    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected);
  });
});
