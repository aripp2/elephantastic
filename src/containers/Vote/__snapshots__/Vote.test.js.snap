// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vote should match the snapshot 1`] = `
<section
  className="vote-section"
>
  <div
    className="voting-btns"
  >
    <button
      className="vote-btn"
      onClick={[Function]}
    >
      LOVE IT!
    </button>
    <button
      className="vote-btn"
      onClick={[Function]}
    >
      NOT SO MUCH
    </button>
  </div>
  <img
    alt="dog"
    className="voteImg"
    src="https://cdn2.thedogapi.com/images/anIClTXyD.jpg"
  />
  <button
    className="fav-btn"
    onClick={[Function]}
  >
    Add Favorite
  </button>
</section>
`;
